### DISCUSSION TITLE
How to deal with output dimensions

### DISCUSSION THREAD

**Discussion by mihlema** [spine-tools/SpineOpt.jl#490] [Ideas]
Hi folks,

just thinking about our model results and what to do with output dimensions. Specifically, what to do with operating points and what to do with stochastic scenarios.

I think for stochastic scenarios, we had at some point decided to go for stochastic maps; however, given we have now also results (not)/overwriting, and analysis times we maybe have to think about when which operation should happen/which index should take precedence.

Similarly, the question is what to do with operating points. It somehow seems to make sense that these would be Arrays in the output, but we don't support Arrays of Timeserieses/Timeseries of Array. So I guess this would then also be a Map parameter?

My first intuition would be to have the following hierarchy, but this is completely open for discussion:
analysis time -> stochastic scenario -> operating point -> time-series values
@Spine-project/spineopt-developers

**Comment by mihlema** [comment]
Related to issue reported here: #489

**Comment by clizbe** [comment]
Have we figured this out? I think I've seen an issue related to stochastic outputs, but don't think it's been solved.

**Comment by Tasqu** [comment]
I don't think this has been "figured out", as this kind of depends on how you model things and what you want to get as output. E.g. `analysis_time`, `stochastic_scenario`, and even `operating_point` can be uninteresting if all we're interested in the realized flow timeseries (total over operating points). However, for debugging purposes, having access to the exact variable values can be helpful.
